numpy/core/src/multiarray/methods.c

```diff
@@ -2563,7 +2565,6 @@ array_setflags(PyArrayObject *self, PyObject *args, PyObject *kwds)
             PyArray_CLEARFLAGS(self, NPY_ARRAY_WRITEABLE);
```

Comment:
Should this clear warn_on_write too? If the caller sets the result of `broadcast_arrays` to have `result.flags.writeable = False`, it seems to me that the warning is not needed.